contracts/ocr2/src/contract.rs

```diff
@@ -259,6 +258,34 @@ pub fn reply(_deps: DepsMut, _env: Env, msg: Reply) -> Result<Response, Contract
     }
 }
 
+// Migrate contract if version is lower than current version
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn migrate(deps: DepsMut, _env: Env, _msg: Empty) -> Result<Response, ContractError> {
+    let stored = cw2::get_contract_version(deps.storage)?;
+
+    // ensure we are migrating from an allowed contract
+    if stored.contract != CONTRACT_NAME {
+        return Err(ContractError::CannotMigrate {
+            previous_contract: stored.contract,
+        });
+    }
+
+    // TODO: could use semver to only allow upgrades:
```

Comment:
Why is this TODO useful here?

The code of the migrate fn is set by the user executing the migration, so as they can add/remove any checks we can't really enforce this, right?